dependencydiff/mapping.go

```diff
@@ -81,9 +81,23 @@ var (
 	}
 )
 
-func toEcosystem(e string) (ecosystem, error) {
-	if ecosystemOSV, found := gitHubToOSV[e]; found {
-		return ecosystemOSV, nil
+func mapDependencyEcosystemNaming(logger *sclog.Logger, deps []dependency) {
+	for i := range deps {
+		if deps[i].Ecosystem == nil {
+			logger.Info("dependency %s has a nil ecosystem", deps[i].Name)
```

Comment:
what happens when we ignore the ecosystem. I mean, what will the final result data be missing?

Reply:
If we return an error when an GH ecosystem cannot be mapped to an OSV one: we'll have a nil `DependencyCheckResult` and an error `cannot find the mapping`. That said, we won't have any valid results if one ecosystem is missing its mapping. My concern is, when GH adds a new ecosystem naming and we haven't had the map updated in Scorecard yet, this API will fail and return nil when that new ecosystem shows in the raw results.

If we ignore the ecosystem mapping missing: we'll still have the final `DependencyCheckResult`, but the ecosystems might have different naming convention - some are mapped to OSV, those missing a mapping are still GH. We'll see the mapping missing information in the log though.

Reply:
having the info in log: do `logger.Info("naming convention mapping not found for dependency xxx")`

Reply:
Info() is for positive things. Any logging for this should be Debug(). We have to decide if we want the check to return an error or not. I'm fine with the entire check failing: we want to know this problem is happening; and I doubt the names will change often.

Reply:
> Info() is for positive things. Any logging for this should be Debug(). 
Re: Info() and Debug(): I see, thanks.

> We have to decide if we want the check to return an error or not. I'm fine with the entire check failing: we want to know this problem is happening; and I doubt the names will change often.

I agree with both: (1) we want to know if this problem is happening and (2) the names might change often, and I haven't gotten a better idea to let the caller know this problem is happening - returning an error solves this and is easy. 

If we keep returning an error for it as it is now, we do need timely and maybe frequent updates on that hashmap if a new GH name comes out; otherwise, the `dependency-diff` will keep failing because of lack of naming convention mapping entries.

Reply:
I don't have better or stronger justifications to make it not return an error - so maybe let's leave it as it is?

I'll create another PR to log it into the logger and fix the typo.

Another issue is, GH has an ecosystem `actions` but OSV doesn't. For this, should I create a map entry `"actions": "actions"`?

Reply:
I'm in favor of returning an error. These names are not going to be changing all the time. They need backward compatibility so they can only *add* names. They support so many already that I think it's a small risk.

Reply:
> I'm in favor of returning an error. These names are not going to be changing all the time. They need backward compatibility so they can only _add_ names. They support so many already that I think it's a small risk.

I see, thanks. Resolving this conversation & closing the PR.